core/frontend/src/tile/RealityTileTree.ts

```diff
@@ -112,6 +113,104 @@ export interface RealityTileTreeParams extends TileTreeParams {
   readonly rootTile: RealityTileParams;
 }
 
+/** The return status for collecting a tile with [[RealityTileCollector.SelectTile]]
+ * @alpha
+ */
+export enum RealityTileCollectionSelectionStatus { Continue, Reject, Accept }
+
+/**  The return status for collecting a set of tiles with [[RealityTileTree.collectTiles.
+ * @alpha
+ */
+export enum RealityTileCollectionStatus { Loading, Success }
+
+/** @alpha */
+export enum RealityTileDrapeStatus { Loading, Success, Error }
+
+/** Base class for a collecting reality tiles from a [[RealityTileTree]].
+ * @alpha
+ */
+export abstract class RealityTileCollector {
+  public acceptedGeometry(): RealityTileGeometry[] {
+    const geometry = new Array<RealityTileGeometry>();
+    for (const accepted of this.accepted) {
+      if (accepted.geometry)
+        geometry.push(accepted.geometry);
+    }
+
+    return geometry;
+  }
+
+  /** Missing (not yet loaded) tiles.  Loading of these tiles can be requested by calling [[RealityTileCollector.requestMissingTiles]].
+   * @internal
+  */
+  public missing = new Set<RealityTile>();
+  /** The accepted tiles. If [[RealityTileTree.collectTiles]] returns [[RealityTileCollectionStatus.Success]] then this contains complete
+   * collection of loaded tiles accepted by [[RealityTileCollector.selectTile]].
+   * @internal
+   */
+  public accepted = new Array<RealityTile>();
+
+  protected constructor(private _tolerance: number, private _range: Range3d, protected _iModelTransform: Transform) {  }
+
+  private _childrenLoading = false;
+
+  /** @internal */
+  public markChildrenLoading() {
+    this._childrenLoading = true;
+  }
+  /** @internal */
+  public get loadingComplete() {
+    return !this._childrenLoading && 0 === this.missing.size;
+  }
+
+  /** Request missing (unloaded) tiles.  Call this method if [[RealityTileTree.collectTiles]] returns [[RealityTileCollectionStatus.Loading]]. */
+  public requestMissingTiles(viewport: Viewport) {
+    IModelApp.tileAdmin.requestTiles(viewport, this.missing);

```

Comment:
`TileAdmin.requestTiles` **replaces** the set of tiles currently requested for use by the viewport - is that what you intend to do?

Reply:
No - Is there a better way to do this?

Reply:
I'm not sure. A bunch of this code looks very similar to `selectTiles`, but happens independently of it - could it be integrated into creating the scene?

Reply:
No - these are independent of scene creation.   The geometry (polyfaces) are extracted from the tiles and used to satisfy some query - draping in this case.

Reply:
Don't you specifically want to drape your line strings on the same geometry that is being displayed though? Could it be simpler and more efficient to have a TileTree that provides both geometry and graphics?

Reply:
For the civil workflow the draped linestrings are displayed in a seperate viewport (they call it profile or something like that) not the same one that is displaying the 3D meshes.   And if they supply that profile viewport when they drape the linestrings then the tile requests should be seperate and correct.  (Nicholas W has already tested this).